pkg/kvstore/etcd.go

```diff
@@ -455,7 +455,7 @@ func (e *etcdClient) renewSession() error {
 
 	e.getLogger().WithField(fieldSession, newSession).Debug("Renewing etcd session")
 
-	if err := e.checkMinVersion(); err != nil {
+	if err := e.checkMinVersion(ctx); err != nil {
```

Comment:
1) This function will block for long periods of time in line 439. This context will be harmful if it is build WithTimeout.